docs/user-guides/retrieval.md

```diff
@@ -0,0 +1,191 @@
+# Retrieval in CLIP-as-service
+`CLIP-as-service` offers us high quality embeddings. Retrieval is one of the most common use cases for embeddings. Retrieval in CLIP-as-service can support indexing a very large dataset(millions/billions) and querying within several million seconds.
+
+In order to implement retrieval we add an extra indexer after enocoder in CLIP-as-service. We use ['AnnLite'](https://github.com/jina-ai/annlite) in this case. 
+
+
+## Fast search in CLIP-as-service
+Index and search are easy in `CLIP-as-service`:
+
+```python
+from clip_client import Client
+
+client = Client('grpc://0.0.0.0:23456')
+
+# index
+r = client.index(
+    [
+        'she smiled, with pain',  # text
+        'https://clip-as-service.jina.ai/_static/favicon.png',  # image
+    ]
+)
+
+# search
+client.search(['smile'])
+```
+
+The results will be like:
+
+```text
+╭───────────────────────────── Documents Summary ─────────────────────────────╮
+│                                                                             │
+│   Length                 1                                                  │
+│   Homogenous Documents   True                                               │
+│   Common Attributes      ('id', 'mime_type', 'text', 'tags', 'embedding')   │
+│   Multimodal dataclass   False                                              │
+│                                                                             │
+╰─────────────────────────────────────────────────────────────────────────────╯
+╭────────────────────── Attributes Summary ───────────────────────╮
+│                                                                 │
+│   Attribute   Data type      #Unique values   Has empty value   │
+│  ─────────────────────────────────────────────────────────────  │
+│   embedding   ('ndarray',)   1                False             │
+│   id          ('str',)       1                False             │
+│   mime_type   ('str',)       1                False             │
+│   tags        ('dict',)      1                False             │
+│   text        ('str',)       1                False             │
+│                                                                 │
+╰─────────────────────────────────────────────────────────────────╯
+```
+
+You don't need to call `client.encode()` explicitly since `client.index()` will handle this for you.
+
+
+Similar to `CLIP-as-service`, you need to prepare a YAML config:
+```yaml
+jtype: Flow
+version: '1'
+with:
+  port: 51000
+executors:
+  - name: encoder
+    uses:
+      jtype: CLIPEncoder
+      metas:
+        py_modules:
+          - clip_server.executors.clip_torch
+  - name: indexer
+  	uses:
+  	  jtype: AnnLiteIndexer
+  	  with:
+  	  	dim: 512
+  	  	data_path: workspace
```

Comment:
`data_path` parameter will be deprecated since it cannot work with multi-shards.